### ISSUE TITLE
Is there an option to add headers to http target request

### ISSUE THREAD

**Issue by isabr85** [DarekDan/NLog.Targets.HTTP#19]
I am looking for a way to add my own headers to http target requests but so far couldn't find any solution for that, is that something achievable in the current solution?

**Comment by DarekDan** [comment]
It is achievable? But not in current state. You can fork and add or file an enhancement request.

Sent from my iPhone

On Jun 2, 2021, at 1:42 PM, isabr85 ***@***.***> wrote:

﻿

I am looking for a way to add my own headers to http target requests but so far couldn't find any solution for that, is that something achievable in the current solution?

**Comment by isabr85** [comment]
How can I file an enhancement request? I already downloaded code base and added the small piece of code that supports headers and it looks good. But I can't create pull request for you because I obviously don't have permission. So how can I file a request?

**Comment by isabr85** [comment]
PR created. #20

**Comment by DarekDan** [comment]
I will review next week.

Sent from my iPhone

On Jun 2, 2021, at 2:59 PM, isabr85 ***@***.***> wrote:

﻿

PR created. #20<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FDarekDan%2FNLog.Targets.HTTP%2Fpull%2F20&data=04%7C01%7C%7C6d086fe691db442fa3b808d926095cf9%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637582643915561158%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=NjzlAxWUVtrfuqO473eX%2BFsQfX7TpxQNYHiMUZkg%2BPo%3D&reserved=0>

**Comment by isabr85** [comment]
I forked this repository and added headers support, the same as I sent in the PR #20 . When I navigating to the controller from my browser, I can see that logs being written into my logs api, but once I am trying to do load tests in order to measure performance (1000 requests per minute for 5 minutes) - no logs been written to my api at all. Why is it working for several manual requests but not working for load tests?

Here is my NLog configuration file:

```
<?xml version="1.0" ?>
<nlog autoReload="true"
      xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <extensions>
    <add assembly="NLog.Targets.Http" />
    <add assembly="NewRelic.LogEnrichers.NLog" />
  </extensions>
  <targets>
    <target name="AsyncInformation" xsi:type="AsyncWrapper" timeToSleepBetweenBatches="1000">
      <target     type="HTTP"
                  name="ws"
                  Method='POST'
                  URL="https://log-api.eu.newrelic.com/log/v1"
                  ContentType='application/json'
                  Accept='application/json'
                  Headers='HeaderKey:HeaderValue'
                  BatchSize='1'
                  MaxQueueSize='10'
                  IgnoreSslErrors='true'
                  FlushBeforeShutdown='true'
                  DefaultConnectionLimit='1000'
                  Expect100Continue='false'
                  UseNagleAlgorithm='true'
                  ConnectTimeout='30000' 
                  InMemoryCompression='true'>
          <layout xsi:type="newrelic-jsonlayout">
            <attribute name="time" layout="${date}" />
            <attribute name="level" layout="${level:upperCase=true}"/>
            <attribute name="message" layout="${message}" />
            <attribute name="host" layout="${machinename}" />
            <attribute name="ActivityId" layout="${activityId}" />
            <attribute name="processId" layout="${processId}" />
            <attribute name="threadid" layout="${threadid}" />
            <attribute name="event-properties" >
              <layout type="JsonLayout" includeAllProperties="true" maxRecursionLimit="0" escapeForwardSlash="true" />
            </attribute>
            <attribute name="exception" layout="${exception}" />
            <attribute name="traceId" layout="${var:traceId}" />
            <attribute name="custom-keys" layout="${var:customKeys}" />
            <attribute name="environment" layout="${environment:ASPNETCORE_ENVIRONMENT}" />
            <attribute name="serviceName" layout="LOGS TESTS" />
            <attribute name="applicationName" layout="LOGS TESTS" />
          </layout>
      </target>
    </target>

  </targets>
  <rules>
    <logger name="*" minlevel="Information" maxlevel="Error" writeTo="AsyncInformation"/>
  </rules>
</nlog>
```

**Comment by DarekDan** [comment]
I am not a fan of string attribute splitting, so I have implemented an alternative approach in version 1.0.9. It allows to add as many headers as required, without having to worry about string escaping (beyond XML encoding of course). Let me know if this can be closed.

**Comment by snakefoot** [comment]
@DarekDan Created #24 since NHttpHeader is missing NLog Layout-support.

**Comment by DarekDan** [comment]
Resolved in 1.0.13 ?
